### PR TITLE
Add IE testing and guard proxies when they are not present

### PR DESCRIPTION
**Pull request by igorT**
We were not testing against IE11, and a few failures had snuck in. We now only create m3 array proxies when we are able to instantiate a Proxy class, similarly to how Ember does it https://github.com/emberjs/ember.js/blob/v3.28.0/packages/@ember/-internals/utils/lib/proxy-utils.ts

### CODE REVIEW

File: tests/dummy/app/adapters/-json-api.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import EmberObject from '@ember/object';
+import fetch from 'fetch';
```
Comment (hjdivad) [inline tests/dummy/app/adapters/-json-api.js:2]: what does this have to do with IE testing?

File: package.json [hunk 2]
```diff
@@ -88,6 +88,7 @@
     "ember-compatibility-helpers": "^1.2.5",
     "ember-disable-prototype-extensions": "^1.1.2",
     "ember-export-application-global": "^2.0.1",
+    "ember-fetch": "^8.1.1",
```
Comment (hjdivad) [inline package.json:91]: what does this have to do with IE testing?

File: tests/dummy/app/adapters/-json-api.js (hunk 1 above)
Comment (spham92) [reply tests/dummy/app/adapters/-json-api.js:2]: I think we need it so it polyfills for IE11?

https://caniuse.com/fetch

File: package.json (hunk 2 above)
Comment (igorT) [reply package.json:91]: Need it for https://github.com/hjdivad/ember-m3/blob/master/tests/dummy/app/adapters/-json-api.js#L5
